uaclient/entitlements/landscape.py

```diff
@@ -0,0 +1,139 @@
+import logging
+import os
+import subprocess
+from typing import Any, Dict, Optional, Tuple
+
+from uaclient import apt, event_logger, exceptions, messages, system, util
+from uaclient.entitlements.base import UAEntitlement
+from uaclient.entitlements.entitlement_status import ApplicationStatus
+
+event = event_logger.get_event_logger()
+
+LANDSCAPE_CLIENT_PACKAGE_NAME = "landscape-client"
+LANDSCAPE_CLIENT_CONFIG_PATH = "/etc/landscape/client.conf"
+LANDSCAPE_CLIENT_CONFIG_PATH_DISABLE_BACKUP = (
+    "/etc/landscape/client.conf.pro-disable-backup"
+)
+
+
+class LandscapeEntitlement(UAEntitlement):
+    name = "landscape"
+    title = "Landscape"
+    description = "Management and administration tool for Ubuntu"
+    help_doc_url = "https://ubuntu.com/landscape"
+
+    def _perform_enable(self, silent: bool = False) -> bool:
+        cmd = ["landscape-config"] + self.extra_args
+        if self.assume_yes and "--silent" not in cmd:
+            cmd += ["--silent"]
+
+        logging.debug(messages.EXECUTING_COMMAND.format(" ".join(cmd)))
+        event.info(
+            util.redact_sensitive_logs(
+                messages.EXECUTING_COMMAND.format(" ".join(cmd))
+            )
+        )
+        try:
+            system.subp(cmd, pipe_stdouterr=self.assume_yes)
+        except exceptions.ProcessExecutionError as e:
+            if self.assume_yes:
+                err_msg = messages.LANDSCAPE_CONFIG_FAILED
+                event.error(
+                    err_msg.msg,
+                    err_msg.name,
+                    service=self.name,
+                    additional_info={
+                        "stdout": e.stdout,
+                        "stderr": e.stderr,
+                    },
+                )
+                event.info(e.stderr)
+                event.info(
+                    messages.ENABLED_FAILED.format(title=self.title).msg
+                )
+            return False
+
+        event.info(messages.ENABLED_TMPL.format(title=self.title))
+        return True
+
+    def _perform_disable(self, silent: bool = False) -> bool:
+        cmd = ["landscape-config", "--disable"]
+        event.info(messages.EXECUTING_COMMAND.format(" ".join(cmd)))
+        try:
+            system.subp(cmd)
+        except exceptions.ProcessExecutionError as e:
+            with util.disable_log_to_console():
+                logging.error(e)
+            event.info(str(e).strip())
+            event.warning(str(e), self.name)
+
+        msg = messages.BACKING_UP_FILE.format(
+            original=LANDSCAPE_CLIENT_CONFIG_PATH,
+            backup=LANDSCAPE_CLIENT_CONFIG_PATH_DISABLE_BACKUP,
```

Comment:
we are creating this backup on disable, right - is there any case where we want to restore this config file when enabling again?

Reply:
We discussed this in the spec and decided not to. We can always add that as an improvement later if we decide that is desired behavior